policybot/pkg/testflakes/testflakes.go

```diff
@@ -0,0 +1,358 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Take in a pr number from path "istio-prow/pr-logs/pull/istio-istio" and examine the pr
+ * for all tests that are run and their results. The results are then written to Spanner.
+ */
+package testflakes
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"strconv"
+	"strings"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/iterator"
+
+	store "istio.io/bots/policybot/pkg/storage"
+)
+
+/*
+ * Pull struct for the structure under refs/pulls in clone-records.json
+ */
+type Pull struct {
+	Number int
+	Author string
+	Sha    string
+}
+
+/*
+ * Cmd struct for Commands object under clone-records.json
+ */
+type Cmnd struct {
+	Command string
+	Output  string
+}
+
+/*
+ * Finished struct to store values for all fields in Finished.json
+ */
+type Finished struct {
+	Timestamp int64
+	Passed    bool
+	Result    string
+}
+
+/*
+ * Clone_Record struct to store values for all fields in clone-records.json
+ */
+type CloneRecord struct {
+	Refs struct {
+		Org       string
+		Repo      string
+		BaseRef   string
+		BaseSha   string
+		Pulls     []Pull
+		PathAlias string
+	}
+	Commands []Cmnd
+	Failed   bool
+}
+
+/*
+ * Tests strut to keep track of the test suite names and the directory for each test runs for the pr.
+ */
+type Tests struct {
+	Name string
+	Prs  []string
+}
+
+/*
+ * Started struct to store values from started.json
+ */
+type Started struct {
+	Timestamp int64
+}
+
+type PrFlakeTest struct {
```

Comment:
This object does not in fact represent a FlakeTest, right? This object lets you test for whether a PR has experienced flakes. So I'd rename this type to FlakeChecker.